### PR TITLE
Users are notified when another user joins the chat

### PR DESCRIPTION
**Pull request by Druddigon**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ io.on('connection', function (socket) {
     socket.emit('new-client', { id: socket.id, username: data.username });
     // Update user list for all clients
     io.emit('update-userlist', chatUserList);
+    socket.broadcast.emit('message', { message: data.username + ' has joined the chat.' });
```
Comment (rongduan-zhu) [inline app.js]: doing this over io.emit because you want all users except for the current user who just joined?

Comment (Druddigon) [inline app.js]: yep

**Comment by rongduan-zhu** [comment]
:+1:

**Comment by nedp** [comment]
:+1:

**Pull request merged**